### DISCUSSION TITLE
Question about the capacity fade in calendar_ageing.py

### DISCUSSION THREAD

**Discussion by Xie-Jiahang** [pybamm-team/PyBaMM#1338] [Q&A] [answered]
Hi, I have a question about how to evaluate capacity degradation in the example of calandar_ageing.py. I'm just a beginner and wish you could solve my problem.

Among the variables which can be output, I searched 'capacity' and get only 'Discharge capacity [A.h]'. But in the calendar_ageing example, it tests only self-discharge, with current function as 0A. So the discharge capacity remains 0 as well. And I saw in this [issue](https://github.com/pybamm-team/PyBaMM/issues/836), it is said the capacity reduces in this example. So I was wondering is 'Discharge capacity [A.h]' the right output value to monitor to show the capacity degradation?

And is it true that we cannot change ['Cell capacity [A.h]'] in default_parameter_values? Because when I changed that, I got pybamm.expression_tree.exceptions.SolverError: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDASolve returned "IDA_CONV_FAIL". Consult IDAS documentation.

**Comment by valentinsulzer** [comment, marked as answer]
We are working on better variables to characterize degradation but progress is a bit slow. For now one quick thing you could look at is the variable "Loss of lithium to negative sei [mol]". If you multiply by Faraday's constant and divide by 3600 that will give you LLI in Ah. 
You can add as a variable to the model by doing
```python3
model.variables["Loss of charge to sei [Ah]"] = model.variables["Loss of lithium to negative sei [mol]"] * model.param.F / 3600
```
right after you define the model.

The "Cell capacity [Ah]" variable is only used to convert from C-rate to Ah (really it should be called "Nominal cell capacity [Ah]"). I don't know why the simulation is failing when you change it.

**Reply by Xie-Jiahang** [reply]
Thank you very much for the advice! I tried to evaluate the loss of capacity as you suggested in SPMe model, setting sei: reaction limited and it worked.

And I want to ask about the resistance measurement, in variables, there is 'Local ECM resistance [Ohm]'; and I think it can be deemed as the internal resistance of battery. So does the average value of this quantity increase in your design when battery degrades?

Besides, I want to include the experiment in a for loop. For example, each experiment is a cycle; and this cycle repeats until the loss of capacity reaches some certain value. Could you please recommend some functions to realize this objective? Since now I try to do this but I don't know whether I have to pass solutions from the last cycle to the parameter values in the new cycle.

Looking forward to your reply, thank you!

**Reply by valentinsulzer** [reply]
Yes, 'Local ECM resistance [Ohm]' is the internal resistance

The experiment takes care of updating things from cycle to cycle for you. Parameters do not change. Sometimes variables change as the cell degrades, for example "active material volume fraction" is a variable that can change if the LAM models are enabled (in the develop branch, not the 0.3.0 release).

Right now there is no way to stop the experiment at 80% capacity fade, again this kind of thing is something we are working on but requires other changes first so might be a while coming.

To change parameter values based on the solution you should be able to define that yourself fairly easily. All the constituent parts for doing that are in the getting started notebooks

**Reply by Xie-Jiahang** [reply]
Thanks for the reply!

Now in my experiment, I just set the times of experiment to be a large number so that hopefully capacity loss can reach 20%. However this process is rather slow, is there a way to accelerate this? For example, add some exponential terms in the growth of 'Loss of lithium to negative sei [mol]'?

Besides, I have found that under different operation conditions, such as different C-rates and CCCV holding voltages, and when these experiments with different conditions have the same capacity loss, the time they take are the same. But it shouldn't be this way, I suppose the right way is that under those different operation conditions, when they reach the same capacity loss, the time it takes should be different, with 'harsh' conditions taking shorter time. So is this something you guys are developing as you mentioned? Or is there a way to solve this problem?

Looking forward to your reply, thank you!

**Reply by nibiyoule** [reply]
我也有相关的问题，pybamm模型如何进行电池的老化仿真。在实验仿真部分我尝试了100次循环测试，发现每一次的结果是相同的，也许我们可以进一步地利用沟通。QQ号：1121062536
I also have a related question about how the PyBamm model can simulate battery aging.In the experimental simulation part, I tried 100 cycles of tests and found that the results were the same each time. Maybe we can use communication further. QQ: 1121062536

**Reply by valentinsulzer** [reply]
You need to include a degradation mechanism such as SEI, plating, or LAM when you define the model: https://pybamm.readthedocs.io/en/latest/source/models/base_models/base_battery_model.html#pybamm.Options
See e.g. https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/calendar_ageing.py

To change how fast the degradation occurs, you need to change the parameters related to the degradation. e.g. for the SEI parameters do `parameter_values.search("SEI")`

**Reply by Mrzhang-hub** [reply]
> Yes, 'Local ECM resistance [Ohm]' is the internal resistance
> 
> The experiment takes care of updating things from cycle to cycle for you. Parameters do not change. Sometimes variables change as the cell degrades, for example "active material volume fraction" is a variable that can change if the LAM models are enabled (in the develop branch, not the 0.3.0 release).
> 
> Right now there is no way to stop the experiment at 80% capacity fade, again this kind of thing is something we are working on but requires other changes first so might be a while coming.
> 
> To change parameter values based on the solution you should be able to define that yourself fairly easily. All the constituent parts for doing that are in the getting started notebooks

As we all know, the internal resistance 'Local ECM resistance [Ohm]' will change while charging or discharging. I wanna ask have you conducted the experiment to find out the relationship between internal resistance and SOC. I have been writing an essay which use the parameters of LGM50_Chen2020 but I need the Internal Resistance-SOC curve which I could not find in that parameter files.

**Comment by aceacedey** [comment]
You may find the following article useful for battery degradation model using PyBaMM:
https://pubs.rsc.org/en/content/articlelanding/2022/CP/D2CP00417H

Code can be found here: 
https://docs.pybamm.org/en/latest/source/examples/notebooks/models/coupled-degradation.html#Modelling-coupled-degradation-mechanisms-in-PyBaMM
